### DISCUSSION TITLE
Create external table with target database's name contains a dot

### DISCUSSION THREAD

**Discussion by minma898** [microsoft/azuredatastudio#15223] [General]
I am trying to create an external table in SQL server using Virtualize Data Wizard. The target database name contains a dot. It causes the errors when trying to mapping to the tables in the database. The database name looks like xxxxx.xxx. Here are errors

105082;Generic ODBC error: [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Could not find server 'xxxxx' in sys.servers. Verify that the correct server name was specified. If necessary, execute the stored procedure sp_addlinkedserver to add the server to sys.servers. .
105082;Generic ODBC error: [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Could not find server 'xxxxx' in sys.servers. Verify that the correct server name was specified. If necessary, execute the stored procedure sp_addlinkedserver to add the server to sys.servers. .
OLE DB provider "MSOLEDBSQL" for linked server "(null)" returned message "Unspecified error".
OLE DB provider "MSOLEDBSQL" for linked server "(null)" returned message "Unspecified error".

I tried to create sane in SSMS using the statement below and I am getting same error. I have put brackets around database name.
 CREATE EXTERNAL TABLE [schema_name].[table_name]
(
    [Column1] [int] NOT NULL,
    [Column2] [int] NOT NULL,
    [Column3] [decimal](18, 5) NOT NULL,
    [Column4] BINARY(8) NOT NULL
)
WITH (LOCATION = N'[xxxxx.xxx].[dbo].[table_name]', DATA_SOURCE = [data_source_name]);

Appreciate any helps with resolving this issue!

**Comment by Charles-Gagnon** [comment]
Following up on this - the issue appears to be in the engine code. I've passed this on to the team for investigation.

The only workaround currently would be to use a server name without a . in it unfortunately.

**Comment by lerchdude** [comment]
This issue has been fixed in SQL Server 2019 CU11, see https://support.microsoft.com/en-us/topic/kb5003870-fix-support-using-the-polybase-sql-server-connector-to-create-external-table-with-remote-database-name-containing-a-period-f920ec30-2e9d-449e-a2af-6586d7c61712.
